concordium-consensus/tests/consensus/ConcordiumTests/KonsensusV1/TransactionProcessingTest.hs

```diff
@@ -128,6 +131,15 @@ instance (MonadReader r m) => MonadReader r (FixedTimeT m) where
     ask = lift ask
     local f (FixedTime k) = FixedTime $ local f . k
 
+newtype NoLoggerT m a = NoLoggerT {runNoLoggerT :: m a}
+    deriving (Functor, Applicative, Monad, MonadIO, MonadReader r, MonadFail)
+
+instance (Monad m) => MonadLogger (NoLoggerT m) where
+    logEvent _ _ _ = return ()
+
+deriving instance (TimeMonad m) => TimeMonad (NoLoggerT m)
+deriving instance (MonadState s m) => MonadState s (NoLoggerT m)
```

Comment:
Why can't you use newtype deriving for this? Why does it need to be this more verbose standalone?